### PR TITLE
Refactor main

### PR DESCRIPTION
**Pull request by alexgallotta**

### CODE REVIEW

File: bottlecap/src/bin/bottlecap/main.rs [hunk 1]
```diff
@@ -46,6 +46,9 @@ use bottlecap::{
 };
 
 use bottlecap::config::{AwsConfig, Config};
+use bottlecap::logs::flusher::Flusher;
```
Comment (astuyve) [inline bottlecap/src/bin/bottlecap/main.rs:49]: can we move these up into the main bottlecap import on line 18?

File: bottlecap/src/metrics/aggregator.rs [hunk 2]
```diff
@@ -129,13 +129,10 @@ impl InsertMetric for CountMetric {
...
 impl Entry {
     fn new_from_metric(id: u64, metric: &Metric) -> Self {
-        let new_metric_value = match metric.first_value() {
-            Ok(value) => value,
-            Err(e) => {
-                error!("failed to parse metric: {:?}", e);
-                0.0
-            }
-        };
+        let new_metric_value = metric.first_value().unwrap_or_else(|e| {
```
Comment (astuyve) [inline bottlecap/src/metrics/aggregator.rs:132]: nice!

**Review by astuyve — APPROVED** [review]

File: bottlecap/src/bin/bottlecap/main.rs (hunk 1 above)
Comment (alexgallotta) [reply bottlecap/src/bin/bottlecap/main.rs]: Oh good catch! There was actually a double import too! I will see if clippy has a rule for that

File: bottlecap/src/bin/bottlecap/main.rs [hunk 3]
```diff
@@ -46,6 +46,8 @@ use bottlecap::{
 };
 
 use bottlecap::config::{AwsConfig, Config};
+use bottlecap::metrics::aggregator::Aggregator;
+use bottlecap::tags::provider::Provider;
```
Comment (duncanista) [inline bottlecap/src/bin/bottlecap/main.rs:50]: Aren't we already importing these?

**Review by duncanista — APPROVED** [review]

File: bottlecap/src/bin/bottlecap/main.rs (hunk 3 above)
Comment (alexgallotta) [reply bottlecap/src/bin/bottlecap/main.rs:50]: yep i cleaned up the imports now

**Pull request merged**